docs/Conceptual/00 - Introduction.md

```diff
@@ -0,0 +1,200 @@
+
+## Solution plane and management plane
+You may have noted that I said "it is usually irrelevant that any given tenant has a parent". There is one respect in which the child tenant *is* aware of the existence of a parent. It needs to be able to *manage its tenancy*. To do so, it will provision, configure, and monitor services through tools provided by the parent that hosts it.
+
+Those tools may also provide means to grant temporary and/or limited access to the child tenant for identities in the parent tenant, in order to deliver support.
+
+We often refer to the divide between the services you *consume in your tenant* and the services you use to *manage your tenant* as the *solution plane* and the *management plane*.
+
+[Solution v. Management Plane](Solution v Management Plane)
+
+Careful conceptual separation of solution plane and management plane is essential.
+
+## Common requirements of a multi-tenanted solution
+Let’s think about a very generalised application. We might say that a typical application has a stimulus/response design like this:
+
+1. Receive a stimulus of some kind (e.g. HTTP request, event, message, timer)
+1. Connect to and authenticate against one or more services within or outside the solution using appropriate credentials, in order to obtain the information required to respond to the stimulus.
+1. Process that information (and optionally respond), possibly connecting to other services within or outside the solution, using appropriate credentials, to update state related to that request.
+
+There are numerous services you can use with that basic pattern – HTTP requests, queues and message busses, event stores, various flavours of databases and storage, caching. The details of these services is outside the scope of this discussion, but they are all consumed using this basic pattern.
+
+In a multi-tenanted solution, you need to layer a notion of "the intended tenant" into your design:
+
+1. Receive a stimulus of some kind, and dispatch it to the appropriate tenant
+1. Connect to and authenticate against one or more services within or outside the tenanted solution using credentials appropriate for the given tenant, in order to obtain the information required to respond to the stimulus.
+1. Process that information (and optionally respond) using code and configuration appropriate for that tenant, possibly connecting to the correct instance of other services within or outside the tenanted solution to update state related to that request, using the instances and credentials appropriate for that tenant.
+
+The way in which this is done is very technology specific, but in essence the environment hosting the tenancy needs to:
+
+1. Identify the intended tenant for the stimulus
+1. Identify the code that is needed to handle the stimulus, and the compute resource to which that stimulus should be directed
+1. Provide a means for that code to find and connect to the services it uses.
```

Comment:
This describes it so well that I don't know what the point of the preceding numbered list is.

Reply:
I think that is a deliberate journey. The first one takes you from people's HTTP bound mental model, and then this second one generalizes and simplifies.